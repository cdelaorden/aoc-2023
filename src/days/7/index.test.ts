import { mapOfOcurrences, parseInput } from '.'

describe('Camel Cards', () => {
  describe('parseInput', () => {
    it('Creates an array of ParsedHands', () => {
      const input = `32T3K 765
      T55J5 684
      KK677 28
      KTJJT 220
      QQQJA 483`
      const result = parseInput(input)
      expect(result).toHaveLength(5)
      expect(result).toMatchSnapshot()
    })
  })
})

describe('mapOfOcurrences', () => {
  it('returns an array of distinct string chars sorted by higher appearence', () => {
    expect(mapOfOcurrences('aaa')).toHaveLength(1)
    const result = mapOfOcurrences('AAABBBCCD')
    expect(result).toHaveLength(4)
    expect(result.at(0)?.at(0)).toBe('A')
    expect(result.at(3)?.at(0)).toBe('D')
  })
})
