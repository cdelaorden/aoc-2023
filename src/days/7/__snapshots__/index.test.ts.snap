// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camel Cards parseInput Creates an array of ParsedHands 1`] = `
[
  {
    "bid": 765,
    "hand": "32T3K",
  },
  {
    "bid": 684,
    "hand": "T55J5",
  },
  {
    "bid": 28,
    "hand": "KK677",
  },
  {
    "bid": 220,
    "hand": "KTJJT",
  },
  {
    "bid": 483,
    "hand": "QQQJA",
  },
]
`;
